kivy/core/window/window_sdl2.py

```diff
@@ -714,7 +714,8 @@ def request_keyboard(self, callback, target, input_type='text'):
 
     def release_keyboard(self, *largs):
         super(WindowSDL, self).release_keyboard(*largs)
-        self._win.hide_keyboard()
+        if self._system_keyboard.widget:
+            self._win.hide_keyboard()
         self._sdl_keyboard = None
```

Comment:
Ok, i read some more source code, and this is a wrong way to do it, system keyboard widget is  None even on android. The example is still good imo, it shows where the problem is. 
The problem is that self._win.hide_keyboard() calls SDL_StopTextInput() in the sdl2 cythonized module. [That function](https://wiki.libsdl.org/SDL_StopTextInput) disables textinput events completely.